lib/elixir/test/elixir/kernel/errors_test.exs

```diff
@@ -37,6 +37,19 @@ defmodule Kernel.ErrorsTest do
       'Foo.:"\#{:bar}"'
   end
 
+  test :invalid_identifier do
+    assert_compile_fail SyntaxError, "nofile:1: unexpected token: foo@", 'foo@'
```

Comment:
Oh, I get it, `foo@` can't be an unexpected token because it's not a valid token.